### PR TITLE
HistoryHandler: fix title in history entry

### LINKED ISSUE #406: Incorrect history entry in browser history when title is a snippet
When title element is used as a snippet, the history in the browser (long click on back button) doesn't work properly in Chrome and Firefox (and probably in Safari as well). 

1. Create a page with a title "A" with ajax link to different page.
2. Navigate using naja to a page with a title "B".
3. The title is correctly set to "B", but in the history, the previous state is also titled "B" instead of "A".

I believe this is caused by changing the title (as any other snippet) before the `pushState` is called. When `pushState` is called, the browser uses the current title value in the history.

This was also an issue in `history.nette.ajax.js` and `nette.ajax.js` libraries (see https://github.com/vojtech-dobes/nette.ajax.js/issues/151). I have forked this library and solved the issue by changing the title only after the `pushState` call. Below are links to respective commmits into those repositories for possible inspiration:
- https://github.com/vojtech-dobes/history.nette.ajax.js/commit/bcd942f73607f3deb82825c60e6c7b87329e8ca2?diff=unified&w=1
- https://github.com/zipper/nette.ajax.js/commit/8b0981e3f1dfc73a12545930c29f81ea13aec8b7?diff=unified&w=1#diff-fce29b22cf71e02cafa4c5d9d05ebf621ddaed2fadaadaea5676027182b2f642R502-R505

There are also some related questions on SO:
- https://stackoverflow.com/questions/26324990/title-of-history-pushstate-is-unsupported-whats-a-good-alternative

### PR DESCRIPTION
**Pull request by jiripudil**
closes #406

### CODE REVIEW

File: src/core/HistoryHandler.ts [hunk 1]
```diff
@@ -48,8 +52,8 @@ export class HistoryHandler extends EventTarget {
 		naja.uiHandler.addEventListener('interaction', this.configureMode.bind(this));
 
 		this.historyAdapter = {
-			replaceState: (state, url) => window.history.replaceState(state, '', url),
-			pushState: (state, url) => window.history.pushState(state, '', url),
+			replaceState: (state, title, url) => window.history.replaceState(state, title, url),
+			pushState: (state, title, url) => window.history.pushState(state, title, url),
```
Comment (filiphlm) [inline src/core/HistoryHandler.ts:56]: We shouldn't do this. The "title" parameter is actually [`unused`](https://html.spec.whatwg.org/multipage/nav-history-apis.html#the-history-interface), Safari is the only browser that has ever supported this, and [that will hopefully end soon](https://github.com/WebKit/WebKit/commit/443a65a4c3a25485b3083c8a92f3be46e71102f2).

This works in Safari as well:

```javascript
history.pushState(state, '', url)
document.title = title
```

Comment (jiripudil) [reply src/core/HistoryHandler.ts:56]: I tested this quite thoroughly in the latest Safari 17.5 (19618.2.12.11.6). While you're right that an empty string works well in a newly pushed state, I've run into an edge case in replaceInitialState: passing an empty string leads to the first history entry having no title at all and displaying the page URL instead.

I'll be more than happy to remove this (again) once Safari 18 is out for some time, but for now, this works more reliably in Safari, without affecting other browsers.

Comment (filiphlm) [reply src/core/HistoryHandler.ts:56]: Oh. I just realized that I don't deal with this (the default/initial history "label" in Safari) anymore. You're right, sorry for the nitpicking 😄

Comment (jiripudil) [reply src/core/HistoryHandler.ts:56]: No problem, thanks for taking a look at the code :)
